tests/integration/targets/ec2_placement_group/tasks/main.yml

```diff
@@ -0,0 +1,218 @@
+          - '"ec2:DescribePlacementGroups" in pg_3_info_result.resource_actions'
+
+    - name: Create a placement group 3 with Partition strategy - Idempotency
+      community.aws.ec2_placement_group:
+        name: '{{ resource_prefix }}-pg3'
+        state: present
+        strategy: partition
+        partition_count: 4
+      register: pg_3_create
+
+    - assert:
+        that:
+          - pg_3_create is not changed
+          - pg_3_create.placement_group.name == '{{ resource_prefix }}-pg3'
+          - pg_3_create.placement_group.state == "available"
+          - '"ec2:CreatePlacementGroup" not in pg_3_create.resource_actions'
+
+    - name: List all placement groups.
+      community.aws.ec2_placement_group_info:
+      register: all_ec2_placement_groups
+
+    - name: Delete a placement group 1
+      community.aws.ec2_placement_group:
+        name: '{{ resource_prefix }}-pg1'
+        state: absent
+      register: pg_1_delete
+
+    - assert:
+        that:
+          - pg_1_delete is changed
+          - '"ec2:DeletePlacementGroup" in pg_1_delete.resource_actions'
+
+    - name: Delete a placement group 1 - Idempotency
+      community.aws.ec2_placement_group:
+        name: '{{ resource_prefix }}-pg1'
+        state: absent
+      register: pg_1_delete
+
+    - assert:
+        that:
+          - pg_1_delete is not changed
+          - '"ec2:DeletePlacementGroup" not in pg_1_delete.resource_actions'
+
+    - name: Delete a placement group 2
+      community.aws.ec2_placement_group:
+        name: '{{ resource_prefix }}-pg2'
+        state: absent
+      register: pg_2_delete
+
+    - assert:
+        that:
+          - pg_2_delete is changed
+          - '"ec2:DeletePlacementGroup" in pg_2_delete.resource_actions'
+
+    - name: Delete a placement group 2 - Idempotency
+      community.aws.ec2_placement_group:
+        name: '{{ resource_prefix }}-pg2'
+        state: absent
+      register: pg_2_delete
+
+    - assert:
+        that:
+          - pg_2_delete is not changed
+          - '"ec2:DeletePlacementGroup" not in pg_2_delete.resource_actions'
+
+    - name: Delete a placement group 3
+      community.aws.ec2_placement_group:
+        name: '{{ resource_prefix }}-pg3'
+        state: absent
+      register: pg_3_delete
+
+    - assert:
+        that:
+          - pg_3_delete is changed
+          - '"ec2:DeletePlacementGroup" in pg_3_delete.resource_actions'
+
+    - name: Delete a placement group 3 - Idempotency
+      community.aws.ec2_placement_group:
+        name: '{{ resource_prefix }}-pg3'
+        state: absent
+      register: pg_3_delete
+
+    - assert:
+        that:
+          - pg_3_delete is not changed
+          - '"ec2:DeletePlacementGroup" not in pg_3_delete.resource_actions'
+
+    - name: List all placement groups.
+      community.aws.ec2_placement_group_info:
+      register: all_ec2_placement_groups
+
+  always:
+
+    - include_tasks: env_cleanup.yml
```

Comment:
You should also add a task to delete all the partition groups inside env_cleanup.yml